### PR TITLE
Patrick Duvall

### PR DESCRIPTION
**Pull request by Patrick-Duvall**
It4

### CODE REVIEW

File: lib/museum.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    @revenue = 0
+  end
+
+  def add_exhibit(exhibit)
+    @exhibits << exhibit
+  end
+
+  def admit(patron)
+    @patrons << patron
+    will_attend = @exhibits.sort{|ex1,ex2|ex2.cost <=> ex1.cost}
```
Comment (s-espinosa) [inline lib/museum.rb]: Intrigued by the choice of a spaceship over a `sort_by`. Not positive if one is more performant than the other, but `sort_by` feels easier to read to me.

File: lib/museum.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    will_attend.each do |exhibit|
+      if recommend_exhibits(patron).include?(exhibit)
+        if patron.spending_money >= exhibit.cost
+          patron.spending_money -= exhibit.cost ; @revenue +=exhibit.cost
```
Comment (s-espinosa) [inline lib/museum.rb]: 👍

File: lib/museum.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  end
+
+  def admit(patron)
+    @patrons << patron
+    will_attend = @exhibits.sort{|ex1,ex2|ex2.cost <=> ex1.cost}
+    will_attend.each do |exhibit|
+      if recommend_exhibits(patron).include?(exhibit)
+        if patron.spending_money >= exhibit.cost
+          patron.spending_money -= exhibit.cost ; @revenue +=exhibit.cost 
+        end
```
Comment (s-espinosa) [inline lib/museum.rb]: Think we could also add a line here to add a patron to a `patrons_of_exhibits` hash for each exhibit they attend to fully complete iteration 4.

File: test/patron_test.rb [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  def test_interests_starts_empty_can_add_to
+    bob = Patron.new('Name',20)
+    assert_equal [], bob.interests
+    bob.add_interest("Dead Sea Scrolls")
+    bob.add_interest("Gems and Minerals")
+    assert_equal ["Dead Sea Scrolls", "Gems and Minerals"], bob.interests
+  end
+
+
+end
```
Comment (s-espinosa) [inline test/patron_test.rb]: Tests look good!

File: test/test_helper.rb [hunk 5]
```diff
@@ -0,0 +1,4 @@
+require "minitest/autorun"
+require "./lib/exhibit"
+require "./lib/patron"
+require "./lib/museum"
```
Comment (s-espinosa) [inline test/test_helper.rb]: Like the test helper and rake file!

**Comment by Patrick-Duvall** [comment]
Yeah I wasnt paying close enough attention and missed that part.

**Pull request closed without merging**